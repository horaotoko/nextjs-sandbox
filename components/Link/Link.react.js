 // Link.react.js
import React, {useEffect} from 'react';
import { useRouter } from 'next/router'
import NextLink from 'next/link';

const Link = () => {
  const router = useRouter();
  const { id } = router?.query;
  let userId = ""

  if (id) {
    userId = id;
  }

  useEffect(() => {
    console.log(id);
  }, [id])

  return (
    <NextLink href="" onClick={() => {console.log('clicked')}}>
      <a href="">ID is {userId}</a>
    </NextLink>
  );
};

export default Link;