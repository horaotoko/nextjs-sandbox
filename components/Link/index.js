import Link from "./Link.react"