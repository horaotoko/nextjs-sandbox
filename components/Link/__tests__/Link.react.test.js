// Link.react.test.js
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import Link from '../Link.react';
import * as Router from 'next/router';
import '@testing-library/jest-dom';

const mockUseRouter = (...params) => {
  return jest.spyOn(Router, 'useRouter').mockImplementation(() => {
    return {
      ...{
        query: {},
        prefetch: jest.fn(),
        // query: { id: 'foo' }
      },
      ...params
    }
  });
};

describe('Link', () => {
  it('render', async () => {
    const router = await mockUseRouter( { query: { id: 'foo' }} );
    console.log(router)
    const { asFragment } = render(
      <Link />
    );

    expect(screen.getByRole('link')).toBeInTheDocument();
    expect(await screen.findByText('ID is foo', {}, {interval: 200})).toBeInTheDocument();

    await waitFor(
      () => {
        // expect(screen.getByRole('link')).toBeInTheDocument();
        // expect(screen.queryByText('ID is bar')).not.toBeInTheDocument();
      },
      {  }
    );

    // expect(asFragment()).toMatchSnapshot();
  });
});