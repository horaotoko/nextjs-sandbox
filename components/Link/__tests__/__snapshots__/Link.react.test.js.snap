// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link render 1`] = `
<DocumentFragment>
  <a
    href=""
  />
</DocumentFragment>
`;
